deployment/src/main/java/io/quarkiverse/quinoa/deployment/PackageManager.java

```diff
@@ -225,61 +215,138 @@ private Command(Map<String, String> envs, String... args) {
             if (frozenLockfile) {
-                return new Command("install", "--frozen-lockfile");
+                return new Command(binary(), "install", "--frozen-lockfile");
             }
-            return new Command("install");
+            return new Command(binary(), "install");
         }
     }
 
     private static class YarnCommands implements Commands {
+        static final String yarn = "yarn";
+
+        @Override
+        public String binary() {
+            return yarn;
+        }
 
         @Override
         public Command install(boolean frozenLockfile) {
             if (frozenLockfile) {
-                return new Command("install", "--frozen-lockfile");
+                return new Command(binary(), "install", "--frozen-lockfile");
             }
-            return new Command("install");
+            return new Command(binary(), "install");
+        }
+    }
+
+    private static class ConfiguredCommands implements Commands {
+        private final Commands detectedCommands;
+        private final PackageManagerCommandsConfig commandsConfig;
+
+        private ConfiguredCommands(Commands detectedCommands, PackageManagerCommandsConfig commandsConfig) {
+            this.detectedCommands = detectedCommands;
+            this.commandsConfig = commandsConfig;
+        }
+
+        @Override
+        public Command install(boolean frozenLockfile) {
+            Command c = detectedCommands.install(frozenLockfile);
+            return new Command(
+                    commandsConfig.installEnv.isEmpty() ? c.envs : commandsConfig.installEnv,
+                    mapToArray(commandsConfig.install).orElse(c.args));
+        }
+
+        @Override
+        public String binary() {
+            return detectedCommands.binary();
+        }
+
+        @Override
+        public Command build(LaunchMode mode) {
+            Command c = detectedCommands.build(mode);
+            return new Command(
+                    commandsConfig.buildEnv.isEmpty() ? c.envs : commandsConfig.buildEnv,
+                    mapToArray(commandsConfig.build).orElse(c.args));
+        }
+
+        @Override
+        public Command test() {
+            Command c = detectedCommands.test();
+            return new Command(
+                    commandsConfig.testEnv.isEmpty() ? c.envs : commandsConfig.testEnv,
+                    mapToArray(commandsConfig.test).orElse(c.args));
+        }
+
+        @Override
+        public Command dev() {
+            Command c = detectedCommands.dev();
+            return new Command(
+                    commandsConfig.devEnv.isEmpty() ? c.envs : commandsConfig.devEnv,
+                    mapToArray(commandsConfig.dev).orElse(c.args));
+        }
+
+        private Optional<String[]> mapToArray(Optional<String> command) {
+            return command.map(new Function<String, String[]>() {
+                @Override
+                public String[] apply(String s) {
+                    return s.split(" ");
```

Comment:
```suggestion
                    return s.split("\\s+");
```

Something like this might me more flexible

Reply:
Maybe this can help: https://stackoverflow.com/questions/1757065/java-splitting-a-comma-separated-string-but-ignoring-commas-in-quotes